library/src/scala/NamedTuple.scala

```diff
@@ -0,0 +1,184 @@
+package scala
+import annotation.experimental
+import compiletime.ops.boolean.*
+
+@experimental
+object NamedTuple:
+
+  opaque type AnyNamedTuple = Any
```

Comment:
I guess this separate definition is needed because `NamedTuple[?, ?]` is illegal, but this definition is very limiting compared to what can be done with a regular `Tuple`:
```scala
def elems(x: Tuple): Int =
  x.size // ok
def wrap(x: Tuple): Tuple.Map[x.type, List] =
  x.map([t] => (x: t) => List(x)) // ok
```
None of these work on `x: AnyNamedTuple` because no extension methods are defined on it.

I can't think of a better way to enable this than writing something like:
```scala
opaque type AnyNamedTuple = Tuple
extension (x: AnyNamedTuple)
  inline def toTuple: Tuple = x
  inline def size: Tuple.Size[Tuple] = toTuple.size
  // ...
  inline def map[F[_]](f: [t] => t => F[t]): AnyNamedTuple =
    toTuple.map(f).asInstanceOf
  // ...
```

and hope that the more precise extension is picked when it is usable.

Reply:
Can you try `AnyNamedTuple = Tuple`? I think I tried that before and it did not work.

Reply:
That doesn't seem to help, somehow all the variants I tried break reduction:
```scala
type Person = (name: String, age: Int)
val xx: NamedTuple.Names[Person] = ("name", "age")
  // error: Found (String, String)  Expected: NamedTuple.Names[Person]
```

Reply:
Hmm. It works for me.